### PR TITLE
changed so that value an be undefined

### PR DESCRIPTION
**Pull request by psvkaushik**

### CODE REVIEW

File: lib/sparql.jison [hunk 1]
```diff
@@ -771,23 +771,23 @@ PathProperties
...
+    | 'END'  Var '='? PathValue?
     {
       if ($4) {
         $$ = { property: 'end', value: { var: $2, value: $4 } };
       } else {
         $$ = { property: 'end', value: { var: $2, value: null } };
       }
     }
-    | 'VIA' Var '='? PathViaValue 
+    | 'VIA' Var '='? PathViaValue
```
Comment (balhoff) [inline lib/sparql.jison]: There should be no equal sign. `VIA` requires a `PathViaValue`, which should be defined as one of Var, PropertyPath, or Pattern.

File: lib/sparql.jison [hunk 2]
```diff
@@ -771,23 +771,23 @@ PathProperties
     ;
 
 PathProperty
-    : 'START' Var '='? PathValue 
+    : 'START' Var '='? PathValue?
```
Comment (balhoff) [inline lib/sparql.jison:775]: Does the jison syntax allow grouping? If there is a PathValue then there must be an equals. So something like `( = PathValue )?`

File: lib/sparql.jison [hunk 3]
```diff
@@ -771,23 +771,23 @@ PathProperties
...
     {
       if ($4) {
         $$ = { property: 'start', value: { var: $2, value: $4 } };
       } else {
         $$ = { property: 'start', value: { var: $2, value: null } };
       }
     }
-    | 'END'  Var '='? PathValue 
+    | 'END'  Var '='? PathValue?
```
Comment (balhoff) [inline lib/sparql.jison:784]: See comment above about grouping.

**Review by balhoff — CHANGES_REQUESTED** [review]

File: lib/sparql.jison (hunk 1 above)
Comment (psvkaushik) [reply lib/sparql.jison]: It got overlooked, but now I removed it.

File: lib/sparql.jison (hunk 2 above)
Comment (psvkaushik) [reply lib/sparql.jison:775]: I thought PathValue should always be present? In that case I can remove the optional for =.

Comment (balhoff) [reply lib/sparql.jison:775]: Only the variable is required, not the optional assignment to a starting value or pattern.

**Pull request merged**